nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -691,15 +691,27 @@ private void executeDML(final ProcessContext context, final ProcessSession sessi
 
                     final Object[] values = currentRecord.getValues();
                     final List<DataType> dataTypes = currentRecord.getSchema().getDataTypes();
-                    List<ColumnDescription> columns = tableSchema.getColumnsAsList();
 
                     for (int i = 0; i < fieldIndexes.size(); i++) {
                         final int currentFieldIndex = fieldIndexes.get(i);
                         Object currentValue = values[currentFieldIndex];
                         final DataType dataType = dataTypes.get(currentFieldIndex);
                         final int fieldSqlType = DataTypeUtils.getSQLTypeValue(dataType);
-                        final ColumnDescription column = columns.get(currentFieldIndex);
-                        int sqlType = column.dataType;
+                        final String fieldName = currentRecord.getSchema().getField(currentFieldIndex).getFieldName();
+                        String columnName = normalizeColumnName(currentRecord.getSchema().getField(currentFieldIndex).getFieldName(), settings.translateFieldNames);
```

Comment:
Very minor, but can you reuse fieldName here?  Also, I see currentRecord.getSchema() called a couple times, could that be pulled into a local variable around line 693?

Reply:
Yeah my bad that's sloppy, I'll change it to reuse local variables for schema, field name, and columns

Reply:
call normalizeColumnName method too many times, maybe store the columnName in a list is better.